caddy/caddymain/run.go

```diff
@@ -98,13 +99,24 @@ func Run() {
 	}
 
 	// Get Caddyfile input
-	caddyfile, err := caddy.LoadCaddyfile(serverType)
+	caddyfileinput, err := caddy.LoadCaddyfile(serverType)
 	if err != nil {
 		mustLogFatalf(err.Error())
 	}
 
+	if validate {
+		justValidate := true
+		_, err := caddy.ValidateAndExecuteDirectives(caddyfileinput, nil, justValidate)
+		if err != nil {
+			mustLogFatalf(err.Error())
```

Comment:
Hmm, just to be safe, use `"%v"` as the first argument, then `err` as the second. (If an error message has "%s" in it, it could create problems I think.) I realize now I didn't do that above and below, as I hadn't thought of this before -- maybe you could do me a favor and change those too? :smile: